src/Sarif/Schemata/sarif-schema.json

```diff
@@ -480,170 +480,173 @@
-                    "items": {
-                        "$ref": "#/definitions/result"
-                    }
-                },
-
-                "taxonomies": {
-                    "description": "An array of reportingDescriptor objects that will be merged with a separate run.",
-                    "type": "array",
-                    "minItems": 0,
-                    "uniqueItems": true,
-                    "default": [],
-                    "items": {
-                        "$ref": "#/definitions/reportingDescriptor"
-                    }
-                },
-
-                "driver": {
-                    "description": "The analysis tool object that will be merged with a separate run.",
-                    "$ref": "#/definitions/toolComponent"
-                },
-
-                "extensions": {
-                    "description": "Tool extensions that will be merged with a separate run.",
-                    "type": "array",
-                    "minItems": 0,
-                    "uniqueItems": true,
-                    "default": [],
-                    "items": {
-                        "$ref": "#/definitions/toolComponent"
-                    }
-                },
-
-                "properties": {
-                    "description": "Key/value pairs that provide additional information about the external properties.",
-                    "$ref": "#/definitions/propertyBag"
-                }
-            },
-            "required": [ "version" ]
+        "properties": {
+          "description": "Key/value pairs that provide additional information about the exception.",
+          "$ref": "#/definitions/propertyBag"
+        }
+      }
+    },
+
+    "externalProperties": {
+      "type": "object",
+      "properties": {
+
+        "schema": {
+          "description": "The URI of the JSON schema corresponding to the version of the external property file format.",
+          "type": "string",
+          "format": "uri"
+        },
+
+        "version": {
+          "description": "The SARIF format version of this external properties object.",
+          "enum": [ "2.0.0-csd.2.beta.2019-02-20" ]
+        },
+
+        "guid": {
+          "description": "A stable, unique identifer for this external properties object, in the form of a GUID.",
+          "type": "string"
+        },
+
+        "runGuid": {
+          "description": "A stable, unique identifer for the run associated with this external properties object, in the form of a GUID.",
+          "type": "string"
+        },
+
+        "conversion": {
+          "description": "A conversion object that will be merged with a separate run.",
+          "$ref": "#/definitions/conversion"
+        },
+
+        "graphs": {
+          "description": "An array of graph objects that will merged with a separate run.",
+          "type": "array",
+          "minItems": 0,
+          "default": [],
+          "uniqueItems": true,
+          "items": {
```

Comment:
according to ballot text, we need to ensure this property is "optional". I wasnt sure how to actually enforce that, isnt everything optional unless we add the "required" property?

If yes, then follow up question - there are items in the ballot text which do not mention the word optional - does that mean they are required? 

text for quick reference:

In the run object:

Change the type of the graphs property from object with graph-valued properties to graph[], unique, minItems: 0, default: empty array.

In the externalProperties object:

Rename graph property to graphs and change its type from object with graph-valued properties to graph[], optional, unique, minItems: 0, default: empty array.